### DISCUSSION TITLE
Sonoff Zigbee Bridge with zigbee2mqtt

### DISCUSSION THREAD

**Discussion by BlueScreenTT** [Koenkk/zigbee2mqtt#10097] [Devices]
Hi

i have installed zigbee2mqtt on my R-PI 4
the addon is starting but i get a loading, please wait screen when trying to access the zigbee2mqtt and devices

what am i doing wrong?

yes i know it is not "supported" but it is experimental
I have ordered the sonoff dongle but it will not arrive until 2022

have anyone gotten this to work?

no matter what i do i can not get it to accept the server or user and password
when i start the addon it makes a backup of the config and creates a new one.

here is the "boot" log

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] socat.sh: executing... 
[19:30:07] INFO: Socat not enabled, marking service as down
[cont-init.d] socat.sh: exited 0.
[cont-init.d] zigbee2mqtt.sh: executing... 
[19:30:08] INFO: MQTT available, fetching server detail ...
[19:30:08] INFO: MQTT server settings not configured, trying to auto-discovering ...
[19:30:09] INFO: Configuring 'mqtt://core-mosquitto:1883' mqtt server
[19:30:09] INFO: MQTT credentials not configured, trying to auto-discovering ...
[19:30:10] INFO: Configuring'addons' mqtt user
[19:30:10] WARNING: Data path /config/zigbee2mqtt not found
[19:30:10] INFO: Adjusting Zigbee2mqtt core yaml config with add-on quirks ...
[cont-init.d] zigbee2mqtt.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[19:30:11] INFO: Handing over control to Zigbee2mqtt Core ...
> zigbee2mqtt@1.22.1 start
> node index.js
Assertion failed
Assertion failed
Zigbee2MQTT:warn  2021-12-08 19:30:29: Using experimental new availability feature
[19:40:21] INFO: Handing over control to Zigbee2mqtt Core ...
> zigbee2mqtt@1.22.1 start
> node index.js
Assertion failed
Assertion failed

\\BlueScreenTT

**Comment by BlueScreenTT** [comment]
i finaly got this 

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] socat.sh: executing... 
[07:28:58] INFO: Socat not enabled, marking service as down
[cont-init.d] socat.sh: exited 0.
[cont-init.d] zigbee2mqtt.sh: executing... 
[07:28:59] INFO: MQTT available, fetching server detail ...
[07:28:59] INFO: Previous config file found, checking backup
[07:28:59] INFO: Creating backup config in '/config/zigbee2mqtt/.configuration.yaml.bk'
[07:28:59] INFO: Adjusting Zigbee2mqtt core yaml config with add-on quirks ...
[cont-init.d] zigbee2mqtt.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[07:29:00] INFO: Handing over control to Zigbee2mqtt Core ...
> zigbee2mqtt@1.22.1 start
> node index.js
Assertion failed
Assertion failed


but i cant get Assertion failed to go away by reading posts on the web

any ideas?

**Comment by Hedda** [comment]
FYI; EZSP support is still in a very experimental phase so you should really post your feedback here -> https://github.com/Koenkk/zigbee-herdsman/issues/319

I am not sure if developer(s) is currently actively working on improving or maintaining that experimental support(?).

Only clear tips is to try upgrade EmberZNet NCP UART EZSP firmware Zigbee firmware version 6.7.9 or later and test dev branch (a.k.a. Edge):

https://github.com/arendst/Tasmota/tree/development/tools/fw_SonoffZigbeeBridge_ezsp

https://github.com/Koenkk/zigbee2mqtt/tree/dev

There looks to be some ezsp related fixes in the zigbee-herdsman dependency that might have not yet made it into Z2M dev/edge yet:

https://github.com/Koenkk/zigbee-herdsman/

Regardless you still want to direct most if not all feedback and issues to subscribers of the WIP discussion in the zigbee-herdsman repo:

https://github.com/Koenkk/zigbee-herdsman/issues/319

Also, please take note that WiFi based Zigbee bridges like ITead Sonoff ZBBridge are generally not recommended with the EZSP protocol. See the warnings about it:

https://www.zigbee2mqtt.io/guide/adapters/#notes

WiFi-based Serial-to-IP bridges are not recommended for Silicon Labs EZSP adapters as the serial protocol does not have enough fault-tolerance to handle packet loss or latency delays that can normally occur over WiFi connections. If cannot use a locally connected USB or UART/GPIO adapter then the recommendation is to use remote adapter that connected via Ethernet (wired) to avoid issues with EZSP caused by WiFi connections.

https://www.zigbee2mqtt.io/advanced/remote-adapter/connect_to_a_remote_adapter.html

WiFi-based Serial-to-IP bridges are not recommended for Silicon Labs EZSP adapters as the serial protocol does not have enough fault-tolerance to handle packet loss or latency delays that can normally occur over WiFi connections.

More warning about Zigbee to WiFi bridges for EZSP is not recommended for production as it does not do well with packed loss:

https://github.com/zigpy/bellows#warning-about-zigbee-to-wifi-bridges

https://www.home-assistant.io/integrations/zha/#warning-about-wi-fi-based-zigbee-to-serial-bridgesgateways

FYI, I also understand that the EFR32 board antenna design of the "SM-011 V1.0" Zigbee module used inside ITead's Sonoff ZBBridge have very high receive sensitivity to electromagnetic interference (EMI) / radio-frequency interference. This means that it will be extra important to keep it away from any other electronic devices or appliances that may generate an electromagnetic field (EM Field) or send our interfering radio signals. That includes the computer that you run your home automation application software on as well as any devices such as for example external harddrives or storage devices, power-supplies, etc., and of course any Wi-Fi access points or routers nearby.

Note! The new ITead Zigbee 3.0 USB Dongle Model: 9888010100045 (Hardware Revision Version 1.3) circuit board share the same type of antenna design so its radio is as well extremely sensitive to signal interference on its receiving end. The current recommendation for using it is to connect it via a long USB extension cable and try to move it as far away from all electronics and other possible sources of signal interference.

PS: Another general tip that always applies when building a Zigbee network is to begin by adding a few good mains-powered Zigbee router devices so that they will make use a stable baseline. Many start by buying three "IKEA TRADFRI Signal Repeater" (E1746).

https://zigbee.blakadder.com/Ikea_E1746.html

https://www.ikea.com/us/en/p/tradfri-signal-repeater-30400407/

https://www.ikea.com/gb/en/p/tradfri-signal-repeater-80424255/

https://www.ikea.com/se/sv/p/tradfri-signalfoerstaerkare-10400408/

**Reply by BlueScreenTT** [reply]
damn that was a lot of good info there :-)
Thank you

I have a mix of 10 sonoff basic with Tasmota and BasicZBR3 non Tasmota scattered around the house so there should be enough routers
i have been using ZHA with out problems until i got the Ikea Trådfri buttons, i dont like the integration and wanted to try Z2M but i guss i will have to wait :-(

1 question.  can i use the tasmota admin addon in homeassistant together with z2m or will there be conflicts ?

**Reply by Hedda** [reply]
> I have a mix of 10 sonoff basic with Tasmota and BasicZBR3 non Tasmota scattered around the house so there should be enough routers

Sonoff Basic ZBR3 uses an older chip (CC2530) with Zigbee Home Automation 1.2 protocol so will not act as a Zigbee router for Zigbee 3.0 devices. Sonoff Basic with Tasmota and other Wi-Fi based Sonoff devices do not use Zigbee no matter if you use Tasmota or not so they will not work as Zigbee routers. Recommend you get some mains-powered Zigbee 3.0 devices to act as Zigbee router devices for IKEA Trådfri Remote as that is a Zigbee 3.0 device. Again the "IKEA TRADFRI Signal Repeater" (E1746) is a good choice to build a baseline Zigbee network with routers that support Zigbee 3.0 specification.

There is no integrations or connections between Tasmota and Z2M so there will be no conflicts there. If you are or will be using Sonoff ZBBridge as a serial stream server to Z2M (or ZHA) then the Tasmota is pass-through mode as thus Z2M (or ZHA) have no awareness of Tasmota. And understand that Z2M can only be used in such a pass-through mode where the serial stream server is just acting as a proxy so Z2M sees the Zigbee coordinator as a local serial port.  If you instead want to the Sonoff ZBBridge with Tasmota in its Tasmote2Zigbee (T2S) mode then Tasmota has full control of the Zigbee coordinator and you can then not connect it to Z2M.
